### DISCUSSION TITLE
create instance

### DISCUSSION THREAD

**Discussion by abbasihamed** [simolus3/drift#1869] [Q&A]
Hi, I use this package for both platforms (web and mobile) and when I want to create an instance from the AppDataBase class, it gives me an error because I have to add some value to the class constructor.
How can I do this without making a mistake?

![image](https://user-images.githubusercontent.com/50572093/170984034-45ce6728-1e17-45fa-80ab-8f9362d21723.png)

![image](https://user-images.githubusercontent.com/50572093/170984260-61d846c9-8d57-4d88-b060-5bd5c362b1b6.png)

**Comment by simolus3** [comment]
> Hi, I use this package for both platforms (web and mobile) and when I want to create an instance from the AppDataBase class, it gives me an error because I have to add some value to the class constructor.

You'd have to use conditional imports to pick the right `QueryExecutor` depending on the current platform. Essentially, create four files: `connection.dart`, `unsupported.dart`, `native.dart` and `web.dart`. In `unsupported.dart`, put:

```dart
import 'package:drift/drift.dart';

QueryExecutor connect() {
  throw UnsupportedError(
      'No suitable database implementation was found on this platform.');
}
```

In `native.dart`, put:

```dart
import 'package:drift/drift.dart';
// ...

QueryExecutor connect() {
  return LazyDatabase(() async {
    // load right path with path_provider
    return NativeDatabase(dbFile);
  });
}
```

In `web.dart`, put

```dart
import 'package:drift/web.dart';

QueryExecutor connect() {
  return WebDatabase('your_app');
}
```

In `connection.dart`, put

```dart
export 'unsupported.dart'
    if (dart.library.html) 'web.dart'
    if (dart.library.ffi) 'native.dart';
```

Then, import `connection.dart` and use `AppDataBase(connect())`.
